tests/tape/interfaces/test_qnode_autograd.py

```diff
@@ -440,7 +440,32 @@ def circuit(a, p):
 
     def test_probability_differentiation(self, dev_name, diff_method, tol):
         """Tests correct output shape and evaluation for a tape
-        with prob and expval outputs"""
+        with a single prob output"""
+
+        dev = qml.device(dev_name, wires=2)
+        x = np.array(0.543, requires_grad=True)
+        y = np.array(-0.654, requires_grad=True)
+
+        @qnode(dev, diff_method=diff_method, interface="autograd")

```

Comment:
It seemed that squeezing is slightly different for each interface (autograd, jax and Torch/TF) as per the previously linked hotfix, could it be worth parametrizing based on the interface?

Reply:
The reason why the QNode must use the ML framework for squeezing, and here we don't, is simply due to where we are with respect to the interface.

* In `QNode.__call__`, we are **outside the interface**. All array/tensor manipulation must be done via the correct ML framework to ensure differentiability is not broken.

* In `tape.parameter_shift`, we are **inside the interface**. Inside the interface, no ML frameworks are used; everything has been unwrapped into a NumPy array. Hence why we can use `np.dot`, and don't have to worry about `torch.matmul`, `tf.tensordot`, etc 🙂

Reply:
Tangent: now that we have a maths module, we should add `qml.math.squeeze`, so that we can remove the `if` statements in `QNode.__call__` to support squeezing in autograd/torch/tf.